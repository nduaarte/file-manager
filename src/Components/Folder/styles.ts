import styled from 'styled-components/native';

export const Container = styled.View`
  
`;

export const Row = styled.View``;

export const InfoContainer = styled.View``;

export const Title = styled.Text``;

export const Quantity = styled.Text``;

export const Percentage = styled.Text``;

