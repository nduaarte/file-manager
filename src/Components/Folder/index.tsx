import React from 'react';
import { Entypo, Feather } from '@expo/vector-icons';

import { Container, Row, InfoContainer, Title, Quantity, Percentage } from './styles';

interface PropsTypes {
  color: string;
  title: string;
  quantity: number;
  percentage: number;
}

const Folder: React.FC<PropsTypes> = ({ color, title, quantity, percentage }) => {
  return(
    <Container>
      <Row>
        <Entypo name='folder' color={color} size={60} />
        <Feather name='more-vertical' size={20} />
      </Row>

      <InfoContainer>
        <Title>{title}</Title>
        <Quantity>{quantity}</Quantity>
        <Percentage>{`${percentage}%`}</Percentage>
      </InfoContainer>
    </Container>
  );
}

export default Folder;