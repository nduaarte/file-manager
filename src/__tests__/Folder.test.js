import React from 'react';
import renderer from 'react-test-renderer';
import Folder from '../Components/Folder';

test('renders correclty', () => {
  const tree = renderer.create(<Folder />).toJSON();
  expect(tree).toMatchSnapshot();
});

