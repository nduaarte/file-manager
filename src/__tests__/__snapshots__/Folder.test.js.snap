// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correclty 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text />
    <Text />
  </View>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
        ]
      }
    />
    <Text
      style={
        Array [
          Object {},
        ]
      }
    />
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      undefined%
    </Text>
  </View>
</View>
`;
